### PR TITLE
FOAM Application Controller

### PR DESCRIPTION
**Pull request by zahnno**
PR Includes: 
- Changed nanos & gen.sh scripts to fix conflicts in build paths. 
- The FOAM Application Controller, which allows you to access components of your foam 
app by using the browser & displaying it as a GUI. 
To access, run the following in the parent directory of foam2:

* Run foam2/./tools/nanos.sh

* Visit http://localhost:8080/static/foam2/src/foam/nanos/controller/index.html

### CODE REVIEW

File: tools/cp.txt [hunk 1]
```diff
@@ -0,0 +1 @@
.../google-http-client-jackson2-1.22.0.jar:/Users/patrick/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.1.3/jackson-core-2.1.3.jar:/Users/patrick/.m2/repository/com/google/guava/guava-jdk5/17.0/guava-jdk5-17.0.jar:/Users/patrick/.m2/repository/org/bouncycastle/bcpkix-jdk15on/1.57/bcpkix-jdk15on-1.57.jar:/Users/patrick/.m2/repository/org/bouncycastle/bcprov-jdk15on/1.57/bcprov-jdk15on-1.57.jar:/Users/patrick/.m2/repository/org/mongodb/mongodb-driver/3.4.2/mongodb-driver-3.4.2.jar:/Users/patrick/.m2/repository/org/mongodb/bson/3.4.2/bson-3.4.2.jar:/Users/patrick/.m2/repository/org/mongodb/mongodb-driver-core/3.4.2/mongodb-driver-core-3.4.2.jar:/Users/patrick/.m2/repository/org/postgresql/postgresql/42.0.0/postgresql-42.0.0.jar:/Users/patrick/.m2/repository/org/java-websocket/Java-WebSocket/1.3.4/Java-WebSocket-1.3.4.jar:/Users/patrick/.m2/repository/com/authy/authy-java/1.1.0/authy-java-1.1.0.jar:/Users/patrick/.m2/repository/org/json/json/20150729/json-20150729.jar:/Users/patrick/.m2/repository/com/google/appengine/appengine-testing/1.9.24/appengine-testing-1.9.24.jar:/Users/patrick/.m2/repository/com/google/appengine/appengine-api-stubs/1.9.24/appengine-api-stubs-1.9.24.jar
```
Comment (kgrgreer) [inline tools/cp.txt]: Should this be checked in?

File: tools/nanos.sh [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#!/bin/sh
 # run from parent directory of foam2
 cd foam2/build/
 mvn dependency:build-classpath -Dmdep.outputFile=cp.txt;
+cp services ../..
```
Comment (kgrgreer) [inline tools/nanos.sh]: Do what nanos does where it does a find and concatenates all services files together.

File: tools/cp.txt (hunk 1 above)
Comment (zahnno) [reply tools/cp.txt]: nope.

Comment (kgrgreer) [reply tools/cp.txt]: Then it should be removed.

**Pull request merged**